### PR TITLE
DEX-1214 New build for replay tests

### PR DESCRIPTION
**Pull request by ngadiyak**

### CODE REVIEW

**Review by JennaWestenra — APPROVED** [review]

File: project/ItTestPlugin.scala [hunk 1]
```diff
@@ -69,7 +69,14 @@ object ItTestPlugin extends AutoPlugin {
...
             (first, second)
           }.getOrElse(DEFAULT_PORT_RANGE)
-          val tests = (Test / definedTests).value
+
+          val tests = Option(System.getenv("REPEATED_CI")) match {
+            case None => (Test / definedTests).value
+            case _ =>
+              val runs = Option(System.getenv("REPEATED_CI_RUNS")).getOrElse("0").toInt
+              val suite = System.getenv("REPEATED_CI_SUITE")
+              for (_ <- 0 to runs) yield (Test / definedTests).value.filter(_.name.contains(suite)).head
```
Comment (vsuharnikov) [inline project/ItTestPlugin.scala:78]: `0 until runs` otherwise we get more runs than it is required

Comment (vsuharnikov) [reply project/ItTestPlugin.scala:78]: Also let's throw an exception if there is no such test. Also there is an opportunity to slightly simplify the code:
```scala
val suiteName = Option(System.getenv("REPEATED_CI_SUITE")).getOrElse(throw new IllegalArgumentException("Specify REPEATED_CI_SUITE"))
val suite = (Test / definedTests).value.find(_.name.contains(suite)).getOrElse(throw new IllegalArgumentException(s"Can't find test *$suiteName*"))
List.fill(runs)(suite)
```

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: project/ItTestPlugin.scala (hunk 1 above)
Comment (ngadiyak) [reply project/ItTestPlugin.scala]: fixed

File: project/ItTestPlugin.scala [hunk 2]
```diff
@@ -69,8 +70,19 @@ object ItTestPlugin extends AutoPlugin {
...
+            case _ =>
+              val runs = Option(System.getenv("REPEATED_CI_RUNS")).getOrElse("0").toInt
+              val suiteName =
+                Option(System.getenv("REPEATED_CI_SUITE")).getOrElse(throw new IllegalArgumentException("Specify REPEATED_CI_SUITE"))
+              val suite = (Test / definedTests).value.find(_.name.contains(suiteName)).getOrElse(throw new IllegalArgumentException(
+                s"Can't find test *$suiteName*"
+              ))
+              List.fill(runs)(suite)
+          }
+          val x = List.fill(3)("foo")
```
Comment (vsuharnikov) [inline project/ItTestPlugin.scala:85]: What is this? ))

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: project/ItTestPlugin.scala (hunk 2 above)
Comment (ngadiyak) [reply project/ItTestPlugin.scala]: it was sth i forget to remove )) sry

**Review by vsuharnikov — APPROVED** [review]

**Pull request merged**